src/GafferModule/ScriptNodeBinding.cpp

```diff
@@ -199,6 +199,7 @@ boost::python::object executionDict( ScriptNodePtr script, NodePtr parent )
 
 std::string serialise( const Node *parent, const Set *filter )
 {
+	IECorePython::ScopedGILLock gilLock;
```

Comment:
I'm pretty sure this should come _after_ the `Py_Initialize()` call. That's where I've put it in #3340.

Reply:
Good catch. Quick question though: This did fix the crash, because python is already initialized. Is it every possible to enter this function without that being the case?

Reply:
The idea is that it's possible to write a pure C++ app that links Gaffer and GafferBindings, and have Python initialised automatically for you when you first call `ScriptNode::serialise()`. So other than linking against GafferBindings, you wouldn't need to use any Python API. I don't think anyone has done that, but that's the reasoning for the initialise...